ddtrace/contrib/gevent/__init__.py

```diff
@@ -5,9 +5,11 @@
 The integration patches the gevent internals to add context management logic.
 
 .. note::
-    If you are using :ref:`ddtrace-run<ddtracerun>` you must set ``DD_GEVENT_PATCH_ALL=true`` and ``ddtrace-run`` will
-    call ``gevent.monkey.patch_all()`` as early as possible in your application to avoid patching conflicts.
-    If you are doing manual instrumentation, you must call ``gevent.monkey.patch_all`` before importing ``ddtrace``.
+    If :ref:`ddtrace-run<ddtracerun>` is being used set ``DD_GEVENT_PATCH_ALL=true`` and
+    ``gevent.monkey.patch_all()`` will be called as early as possible in the application
+    to avoid patching conflicts.
+    If ``ddtrace-run`` is not being used then be sure to call ``gevent.monkey.patch_all``
+    before importing ``ddtrace``.
```

Comment:
Should we also add a note that in the non-`ddtrace-run` case users should run `ddtrace.monkey.patch[_all]` after gevent monkey patching has occurred?

Reply:
We actually need gevent monkey patching to be done prior to even importing ddtrace. The change that inspired this doc update was #2546 which would start the writer thread as soon as ddtrace is imported but I think we can start recommending this ahead of time so that we don't have to keep working around gevent.

Reply:
Right, I agree with the need for this note (import ddtrace after gevent monkey patching). If they don't import ddtrace until after monkey patching, they wouldn't be running ddtrace patching until after gevent monkey patching. Still, I thought we could clarify the need to run ddtrace patching after gevent monkey patching (even though the rule to not import ddtrace before gevent patching ensures this).

Reply:
Ah gotcha! I will also mention patching to be explicit.